From: Jiri Bohac <jbohac@suse.cz>
Subject: bonding: minimize race window on miimon rearming
Patch-mainline: no
References: bnc#602969

This is a temporary solution to minimize the chances of a race condition in the mii-monitor 
re-arming code.

I have been working with upstream on a proper fix, but the latest proposal still has
unresolved problems:
	http://kerneltrap.org/mailarchive/linux-netdev/2010/9/16/6285356
Until these problems are found and fixed, this patch should minimize the chances of hitting 
the problem (verified by Dell in bnc#602969 Comment #25).


diff --git a/drivers/net/bonding/bond_main.c b/drivers/net/bonding/bond_main.c
index 0b4baa0..c0fb4bb 100644
--- a/drivers/net/bonding/bond_main.c
+++ b/drivers/net/bonding/bond_main.c
@@ -2461,7 +2461,7 @@ void bond_mii_monitor(struct work_struct *work)
 	}
 
 re_arm:
-	if (bond->params.miimon)
+	if (bond->params.miimon && !bond->kill_timers)
 		queue_delayed_work(bond->wq, &bond->mii_work,
 				   msecs_to_jiffies(bond->params.miimon));
 out:
